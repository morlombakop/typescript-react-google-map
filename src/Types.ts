export interface ILocation {
  bearing: number
  latitude: number
  longitude: number
}

export interface IDriver {
  driver_id: string
  location: ILocation
}

export interface IBooking {
  pickup_eta: number
  drivers: IDriver[]
}

export interface ITravelMatrix {
  value: number
  text: string
}

export interface IGoogleMapProps {
  drivers: IDriver[]
}

export interface IRangeSliderProps {
  sliderValue: number;
  onSlideComplete: (sliderValue: number) => void
}
