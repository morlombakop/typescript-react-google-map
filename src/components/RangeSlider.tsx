import * as React from 'react'
import Slider, { createSliderWithTooltip } from 'rc-slider'
import styled from 'styled-components'
import { IRangeSliderProps } from '../Types'

const RangeSlider: React.FC<IRangeSliderProps> = ({ onSlideComplete, sliderValue }) => {
  const SliderWithTooltip = createSliderWithTooltip(Slider)
  const Container = styled.div`
    position: absolute;
    bottom: 20px;
    left: 0;
    width: calc(100% - 30px);
    padding: 0 15px;
  `
  return (
    <Container data-testid="range-slider-container">
      <SliderWithTooltip
        defaultValue={sliderValue}
        step={1}
        min={1}
        max={50}
        onAfterChange={onSlideComplete}
        railStyle={{ backgroundColor: '#d3d3d3', height: 7 }}
        trackStyle={{ backgroundColor: '#4caf50', height: 7 }}
        handleStyle={{
          borderColor: '#4caf50',
          height: 28,
          width: 28,
          marginLeft: -14,
          marginTop: -9,
          backgroundColor: '#4caf50',
        }}
        data-testid="input-range-slider"
      />
    </Container>
  )
}

export default RangeSlider
