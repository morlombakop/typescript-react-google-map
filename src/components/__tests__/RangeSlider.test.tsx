import 'jest-dom/extend-expect'
import 'jest-styled-components'
import * as React from 'react'
import { cleanup, render } from '@testing-library/react'
import RangeSlider from '../RangeSlider'
import { IRangeSliderProps } from '../../Types'

afterAll(cleanup)

describe('RangeSlider component tests', () => {
  const props: IRangeSliderProps = {
    sliderValue: 10,
    onSlideComplete: (sliderValue: number) => {},
  }

  const { getByTestId } = render(<RangeSlider {...props} />)
  const rangeContainer = getByTestId('range-slider-container')

  test('RangeSlider component renders properly', () => {
    expect(rangeContainer).toMatchSnapshot()
    expect(rangeContainer).toBeInstanceOf(HTMLDivElement)
    expect(rangeContainer.childElementCount).toBe(1)
    expect(rangeContainer).toHaveStyle(`
      position: absolute;
      bottom: 20px;
      left: 0;
      width: calc(100% - 30px);
      padding: 0 15px;
    `)
    expect(rangeContainer.firstChild).toBeInstanceOf(HTMLElement)
  })
})
