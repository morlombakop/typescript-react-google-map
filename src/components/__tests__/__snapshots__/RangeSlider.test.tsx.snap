// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider component tests RangeSlider component renders properly 1`] = `
.c0 {
  position: absolute;
  bottom: 20px;
  left: 0;
  width: calc(100% - 30px);
  padding: 0 15px;
}

<div
  class="c0"
  data-testid="range-slider-container"
>
  <div
    class="rc-slider"
  >
    <div
      class="rc-slider-rail"
      style="background-color: rgb(211, 211, 211); height: 7px;"
    />
    <div
      class="rc-slider-track"
      style="background-color: rgb(76, 175, 80); height: 7px; left: 0%; width: 18.367346938775512%;"
    />
    <div
      class="rc-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="50"
      aria-valuemin="1"
      aria-valuenow="10"
      class="rc-slider-handle"
      role="slider"
      style="border-color: #4caf50; height: 28px; width: 28px; margin-left: -14px; margin-top: -9px; background-color: rgb(76, 175, 80); left: 18.367346938775512%;"
      tabindex="0"
    />
    <div
      class="rc-slider-mark"
    />
  </div>
</div>
`;
